apps/jupyter/extensions/notebook-runner/src/task-importer.ts

```diff
@@ -28,11 +28,14 @@ const extractFolder = async (
   const folderPath = prefix.endsWith("/") ? prefix : `${prefix}/`;
 
   for (const [path, file] of Object.entries(zip.files)) {
-    const relativePath = path.substring(folderPath.length);
-    if (relativePath) {
-      const blob = await file.async("blob");
-      files.set(relativePath, blob);
-      console.log(`extracted from ${relativePath} from ${folderPath}`);
+    // Only process files within the specified folder
+    if (path.startsWith(folderPath) && !file.dir) {
+      const relativePath = path.substring(folderPath.length);
+      if (relativePath) {
```

Comment:
When can this be falsey? I don't think this should ever happen, right?
The only way I see this could happen is if `path` == `folderPath` which could only happen if a path to a file is provided to the function instead a path to a folder. Should we just throw if this happens and report the wrong values? (guard clause)